dynamic_listings/python/default/collisiondetection/circle_polygon.py

```diff
@@ -8,18 +8,15 @@ def circle_polygon(poly: Polygon, circ:Circle) -> bool:
         # If we get to the end, we wrap around j
         if j == len(poly.vertices):
             j = 0
-        temp_line: Line = Line.fromPoints(poly.vertices[i], poly.vertices[j])
+        temp_line = Line.from_points(poly.vertices[i], poly.vertices[j])
         # In case we find a hit, we already know there is a collision
         if line_circle_collision(circ, temp_line):
             return True
+        
+    return polygon_point_collision(poly, circ.center) or circle_point_collision(circ, poly.vertices[0])
     # Now Let's check for cases "A" and "B"
-    if polygon_point(poly, circ.center):
-        # If the center is inside the polygon, we have a collision
-        return True
-    # Now let's check for the rare edge-case: if this case happens, all the vertices
+    # A: If the center is inside the polygon, we have a collision
+    # B: if this rare edge-case happens, all the vertices
     # are inside the circle, so we can only check one of them
-    if circle_point_collision(circ, poly.vertices[0]):
-        # If any vertex is inside the circle, we have a collision, so we check the first
-        return True
-    # If none of the checks above returned, we don't have a collision (case D)
-    return False
+    # If any vertex is inside the circle, we have a collision, so we check the first
+    # If none of the checks above returned, we don't have a collision (case D)
```

Comment:
I know this is not really "pythonic" but I think it may be clearer to leave the single checks "expanded", instead of grouping them into a single return statement. Throwing everything into a single return statement makes the comments a bit confusing.